### ISSUE TITLE
Normal highlight not defined when compiled with `--disable-gui` (E411)

### ISSUE THREAD

**Issue by blueyed** [vim/vim#4072]
When configuring Vim with `--disable-gui`, `:hi Normal` will result in "E411: highlight group not found: Normal" with `vim -u NONE`.

I think it should be defined always.

With `--enable-gui` it defaults to "cleared".

**Comment by blueyed** [comment]
The following patch fixes it:
```diff
diff --git i/src/syntax.c w/src/syntax.c
index 16c75f27d..a59d231dc 100644
--- i/src/syntax.c
+++ w/src/syntax.c
@@ -6954,6 +6954,8 @@ static char *(highlight_init_light[]) = {
         "MatchParen term=reverse ctermbg=Cyan guibg=Cyan"),
 #ifdef FEAT_GUI
     "Normal gui=NONE",
+#else
+    "Normal cterm=NONE",
 #endif
 #ifdef FEAT_TERMINAL
     CENT("StatusLineTerm term=reverse,bold cterm=bold ctermfg=White ctermbg=DarkGreen",
@@ -7050,6 +7052,8 @@ static char *(highlight_init_dark[]) = {
 #endif
 #ifdef FEAT_GUI
     "Normal gui=NONE",
+#else
+    "Normal cterm=NONE",
 #endif
 #ifdef FEAT_TERMINAL
     CENT("StatusLineTerm term=reverse,bold cterm=bold ctermfg=Black ctermbg=LightGreen",
```

Likely better:
```diff
 src/syntax.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git i/src/syntax.c w/src/syntax.c
index 16c75f27d..41742c0cd 100644
--- i/src/syntax.c
+++ w/src/syntax.c
@@ -6952,9 +6952,8 @@ static char *(highlight_init_light[]) = {
 #endif
     CENT("MatchParen term=reverse ctermbg=Cyan",
 	 "MatchParen term=reverse ctermbg=Cyan guibg=Cyan"),
-#ifdef FEAT_GUI
-    "Normal gui=NONE",
-#endif
+    CENT("Normal cterm=NONE",
+	 "Normal gui=NONE"),
 #ifdef FEAT_TERMINAL
     CENT("StatusLineTerm term=reverse,bold cterm=bold ctermfg=White ctermbg=DarkGreen",
 	 "StatusLineTerm term=reverse,bold cterm=bold ctermfg=White ctermbg=DarkGreen gui=bold guifg=bg guibg=DarkGreen"),
@@ -7048,9 +7047,8 @@ static char *(highlight_init_dark[]) = {
     CENT("Conceal ctermbg=DarkGrey ctermfg=LightGrey",
 	 "Conceal ctermbg=DarkGrey ctermfg=LightGrey guibg=DarkGrey guifg=LightGrey"),
 #endif
-#ifdef FEAT_GUI
-    "Normal gui=NONE",
-#endif
+    CENT("Normal cterm=NONE",
+	 "Normal gui=NONE"),
 #ifdef FEAT_TERMINAL
     CENT("StatusLineTerm term=reverse,bold cterm=bold ctermfg=Black ctermbg=LightGreen",
 	 "StatusLineTerm term=reverse,bold cterm=bold ctermfg=Black ctermbg=LightGreen gui=bold guifg=bg guibg=LightGreen"),
```

**Comment by brammool** [comment]
Daniel Hahler wrote:

> The following patch fixes it:
> ```diff
> diff --git i/src/syntax.c w/src/syntax.c
> index 16c75f27d..a59d231dc 100644
> --- i/src/syntax.c
> +++ w/src/syntax.c
> @@ -6954,6 +6954,8 @@ static char *(highlight_init_light[]) = {
>          "MatchParen term=reverse ctermbg=Cyan guibg=Cyan"),
>  #ifdef FEAT_GUI
>      "Normal gui=NONE",
> +#else
> +    "Normal cterm=NONE",
>  #endif
>  #ifdef FEAT_TERMINAL
>      CENT("StatusLineTerm term=reverse,bold cterm=bold ctermfg=White ctermbg=DarkGreen",
> @@ -7050,6 +7052,8 @@ static char *(highlight_init_dark[]) = {
>  #endif
>  #ifdef FEAT_GUI
>      "Normal gui=NONE",
> +#else
> +    "Normal cterm=NONE",
>  #endif
>  #ifdef FEAT_TERMINAL
>      CENT("StatusLineTerm term=reverse,bold cterm=bold ctermfg=Black ctermbg=LightGreen",
> ```

Should not be difficult to write a test for this.

-- 
hundred-and-one symptoms of being an internet addict:
39. You move into a new house and decide to Netscape before you landscape.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by tonymec** [comment]
I confirm the problem in vi (Tiny Vim) 8.1.1002, my only build without GUI at the moment. I haven't tried applying the patch, but of course, after typing
```vim
:hi Normal cterm=NONE
:hi Normal
```
the result is
```
Normal         xxx cleared
```
(with no change in the display colours) which is the expected result.

For comparison, in my usual Big gvim with GTK2 GUI, when invoked with `-V1 -u DEFAULTS`, after
```vim
:hi Normal
```
the result is (in Console mode)
```
Normal         xxx cleared
```
or (in the GUI)
```
Normal         xxx font=Monospace 10
```
both of which are unsurprising and, I suppose, expected.

Best regards,
Tony.

**Comment by tonymec** [comment]
P.S. Are monochrome monitors still supported? If they are, I suppose the default ought to be the equivalent of
```vim
if has('gui')
    if has('win32')
        hi Normal gui=NONE
    else " non-Windows gui-enabled builds can also be used in Console mode
        hi Normal term=NONE cterm=NONE gui=NONE
else
    hi Normal term=NONE cterm=NONE
endif

**Comment by chrisbra** [comment]
@blueyed can you provide a proper patch with a test?

**Comment by blueyed** [comment]
@chrisbra 
Sorry, no time for this anytime soon.

**Comment by chrisbra** [comment]
Okay, I made a patch: https://github.com/chrisbra/vim/commit/5e3b22bf1818bfc3bf5bed303efc0bb8b094fb3d.patch

Some things to note:

1) I moved the definition of the Normal highlighting group to `highlight_init_both`, as the definition is the same, so it seems sensible to have this definition only once
2) wrote an initial test. The test case must be the first to be called inside test_highlighting.vim. I suppose this is because using the `dump()` feature will define the Normal group anyhow (at least for some reason after the `Test_cursorline...()` functions, the new test does not fail anymore, so that is how I explain it 😕 )
3) slightly amended the documentation and moved the definition of `hl-CursorLineNr` to where it belongs, right after `hl-CursorLine` and added an entry `hl-lCursor`. That one is only briefly mentioned in the source and only briefly in `'guicursor'`, so I added some references to the related options.

According to the documentation `:h hl-Cursor` could be expected to always exists. However currently it is only defined for gui-enabled builds. I did not change that.
